### PR TITLE
Prevent water blocks being breakable

### PR DESCRIPTION
**Pull request by Hopson97**
BEFORE
image

AFTER
image

Also rename "meshType" to just "type" for voxels, as it more reflects what it is

### CODE REVIEW

**Comment by b0guslaw** [comment]
The way I understand it this PR does two things: Make water blocks unbreakable and prevent the selection box to render on water on blocks. The commit message does not reflect this appropriately, I recommend splitting it into two commits, or fix the commit message so it reflects both changes. 

Also can you explain why meshType was renamed to type, whereas meshStyle did not undergo that change?

**Comment by Hopson97** [comment]
Agreed about the commit message.

As for the renaming, the original name `meshType` was wrong, `type` makes more sense as it says what sort physical state the voxel is, for example a "solid" voxel, a "gas" voxel, or a "fluid" voxel. These have very little to do with the actual mesh of the voxel.

MeshStyle is correct as it says the style of the mesh. For example, blocks would have a "block" mesh style, whereas things like flowers and such would have an "X" shaped mesh style.

File: src/client/client.cpp [hunk 1]
```diff
@@ -48,6 +48,10 @@ void renderChunks(const std::vector<ChunkDrawable> &chunks,
         }
     }
 }
+
+bool blockIsBreakable(VoxelType voxelType) {
```
Comment (rehwinkel) [inline src/client/client.cpp]: I'd change this function to something like isBlockSelectable and returning False if the voxelType is any liquid. blockIsBreakable, in my opinion, means if the block is breakable by a player in survival mode (e.g. Bedrock in minecraft... It's selectable but not breakable.)

Comment (Hopson97) [reply src/client/client.cpp]: Agreed, have done this now

**Comment by rehwinkel** [comment]
I think it's good to merge!

**Pull request merged**